### ISSUE TITLE
Consider using multiprocessing workers

### ISSUE THREAD

**Issue by ThomasDelteil** [zackchase/mxnet-the-straight-dope#456]
A lot of people are using these excellent tutorials as a base to write their gluon code. I think it could be good to have the dataloaders by default using multiprocessing for loading:
```
import multiprocessing
...
NUM_WORKERS = multiprocessing.cpu_count()
...
train_data = gluon.data.DataLoader(gluon.data.vision.MNIST(train=True, transform=transform),
                                      batch_size, shuffle=True, num_workers=NUM_WORKERS)
test_data = gluon.data.DataLoader(gluon.data.vision.MNIST(train=False, transform=transform),
                                     batch_size, shuffle=False, num_workers=NUM_WORKERS)
```

In my setup I got a **2x** performance increase using 8 workers, on this tutorial https://github.com/zackchase/mxnet-the-straight-dope/blob/master/chapter04_convolutional-neural-networks/cnn-gluon.ipynb.

Modifying the code to use a non-blocking accuracy and loss and got a **7x** performance increase. see here: https://s3.us-east-2.amazonaws.com/tdelteil-test-mxnet/cnn-gluon.ipynb

**Comment by QiXuanWang** [comment]
Don't know it even exists. Thanks for sharing.

**Comment by zackchase** [comment]
Thanks for the suggestion! That's a good point. Let's think about when to introduce it. I like to have at least one point in the book when everything is explained clearly. Maybe we could introduce the multiprocessing the first time we start using vision datasets?

**Comment by QiXuanWang** [comment]
Don't remember if you have "load/save" introduced in data api chapter. But could combine them together. I think it's kind of useful to know nd.load/nd.save also works. Maybe also combine with parameter save/load?
